federation/src/main/scala/caliban/federation/Federation.scala

```diff
@@ -96,11 +96,13 @@ trait Federation {
       _fieldSet: FieldSet = FieldSet("")
     )
 
+    val withSDL = original.withAdditionalTypes(resolvers.map(_.toType))
```

Comment:
We add these only here to avoid polluting the original graph since these new types don't have any utility since they couldn't be accessed through a normal query anyway.